wgpu/backends/wgpu_native/_api.py

```diff
@@ -2731,10 +2719,69 @@ def begin_render_pass(
 
         # H: WGPURenderPassEncoder f(WGPUCommandEncoder commandEncoder, WGPURenderPassDescriptor const * descriptor)
         raw_encoder = libf.wgpuCommandEncoderBeginRenderPass(self._internal, struct)
-        encoder = GPURenderPassEncoder(label, raw_encoder, self)
+        encoder = GPURenderPassEncoder(label, raw_encoder, self._device)
         encoder._objects_to_keep_alive = objects_to_keep_alive
         return encoder
 
+    # Pulled out from create_render_pass because it was too large.
+    def _create_c_render_pass_depth_stencil_attachment(self, ds_attachment):
```

Comment:
FYI.  I really wanted to put _create_render_pass_color_attachment() into a separate function.  However apipatcher.py::StructValidationChecker complained that it couldn't find `check_struct("Color", clear_value)` in `begin_render_pass()`.  

I could only get away with refactoring because the depth_stencil_attachment has no internal structures.

I wish there were a way that the validation checker could more easily support refactoring.

Reply:
I recently ran into a similar problem with the sync/async versions of a method to both call e.g. `_request_device()`. For these particular cases I could easily implement a fix/hack to see that the call was delegated to a method of the same name, but prefixed with an underscore.

Providing a mechanism to allow delegation mid-function would be rather complex. A quick fix could be would be to tell the check checked that its ok, by adding a commented version of the call, with a note to other humans to tell them that method X must do that check.